icu4c/source/i18n/uspoof_impl.cpp

```diff
@@ -945,7 +945,7 @@ uspoof_swap(const UDataSwapper *ds, const void *inData, int32_t length, void *ou
     uint32_t magic = ds->readUInt32(spoofDH->fMagic);
     ds->writeUInt32((uint32_t *)&outputDH->fMagic, magic);
 
-    if (outputDH->fFormatVersion != spoofDH->fFormatVersion) {
+    if (&outputDH->fFormatVersion != &spoofDH->fFormatVersion) {
```

Comment:
This probably works; not entirely sure without reproducing the warning.

In the ticket discussion I had proposed
```
if (inBytes != outBytes) {
```
instead, like on line 921. What do you think?

Reply:
sure. no strong opinion. I updated it with your suggestion.